helix-core/src/main/java/org/apache/helix/controller/stages/MessageGenerationPhase.java

```diff
@@ -371,6 +401,14 @@ private boolean shouldCleanUpPendingMessage(Message pendingMsg, String currentSt
     }
   }
 
+  private boolean shouldCleanUpStaleMessage(Message staleMsg, Long currentStateTransitionEndTime) {
+    if (staleMsg == null) {
```

Comment:
Can staleMsg be null?

Reply:
Normally it shouldn't. But I'm not 100% sure. I saw pendingMessage has similar check. Want to keep it for safe.

Reply:
I personally prefer to fail early. If it should not, then we should not check. The concern is that overprotected code will hide potential issues.
For example, if some bugs make it always null, then you will always get false as a result. But it would be better that we get an NPE, then the bug will be identified the first time.